public/jsx/components/card-form.jsx

```diff
@@ -138,135 +110,122 @@ module.exports = React.createClass({
           url: '/api/braintree/subscriptions/',
           method: 'post',
           dataType: 'json',
-        }).then(function() {
+          context: that,
+        }).done(function() {
           console.log('Successfully subscribed + completed payment');
           router.transitionTo('complete');
+        }).fail(function($xhr) {
+          var errors = this.processApiErrors($xhr.responseJSON);
+          this.setState({
+            isSubmitting: false,
+            errors: errors,
+          });
         });
       }
     });
   },
 
-  // Just a convenience mapping for cards from card-validator
-  // to shorted classes used in CSS.
-  cardTypeMap: {
-    'american-express': 'amex',
-    'diners-club': 'diners',
-    'master-card': 'mastercard',
-  },
-
-  render: function() {
-    var fieldList = [];
-    var that = this;
-    var allValid = true;
-    var isButtonDisabled;
-    var detectedCard = null;
-
-    var { fields, ...formAttrs } = this.props;
-
-    fields.map(function(field, index) {
-      var showFieldError = false;
-
-      // This uses ES7 'destructuring assignments' to
-      // pass every key *not* starting with '...' to
-      // vars and the remaining key value pairs are left
-      // to be passed into the element with {...attrs}
-      // helps a lot to DRY things up.
-      var { label, placeholder, validator,
-            classNames, pattern, error, ...attrs } = field;
-
-      var val = that.state[field.id];
-      var cardClassName;
-      var fieldClass;
-      // Operate on a copy of the classNames list.
-      var fieldClassNames = classNames &&
-        classNames.slice ? classNames.slice(0) : [];
-
-      // Validate the value
-      if (val && validator) {
-        // We strip out the '_' added to the value by react-masked-input.
-        var valData = CardValidator[validator](val.replace(/_/g, ''));
-        var isValid = valData.isValid === true;
-        var isPotentiallyValid = valData.isPotentiallyValid;
-
-        // Show an error when we know the field is truly invalid
-        showFieldError = !isValid && !isPotentiallyValid;
-
-        if (!isValid) {
-          fieldClassNames.push('invalid');
-          allValid = false;
+  processApiErrors: function(errors) {
+    errors = errors || {};
+    var newErrors = {
+      fields: {},
+      generic: [],
+    };
+    if (errors.error_response && errors.error_response.braintree) {
+      var apiErrors = errors.error_response.braintree;
+      // Iterate over the error object and create a new data
+      // structure keyed by field or otherwise push onto
+      // a list of generic errors.
+      Object.keys(apiErrors).forEach(function(key) {
+        var errorList = apiErrors[key];
+        for (var i = 0; i < errorList.length; i += 1) {
```

Comment:
I'll look into that. Still getting used to all the new possibilities. It's a brave new world!